### PR TITLE
adding tests

### PR DESCRIPTION
**Pull request by ds2643**
unit tests complete... results in results.txt file

### CODE REVIEW

File: tests/unit.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# This file contains a series of unit tests that compare expected and actual javascript code generated by pyd3xter
+# instructions:
+# install pytest: http://pytest.org/latest/
+# $> python -m pytest -v unit.py
+# TODO: refactor to streamline js comparisons; consider edge cases, syntactic curbballs
+# note: expected_js field hand-written
+
+# BUG: inproper spacing
```
Comment (NathanEpstein) [inline tests/unit.py]: These first 8 lines are basically docs. For consistency, I would move them to the README. Some minor notes.
- TODO on line 5 - inline todos are usually for specific issues, this basically just says "make the tests better". 
- line5: 'syntactic curbballs' should be 'curve balls' (typo)
- line8: should be 'improper spacing' (typo)

File: tests/unit.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	test_input = [1,2,3,4,5]
+	chart_type = 'hist'
+	expected_js = strip_space("new D3xter({}).hist([1,2,3,4,5]); \n")
+	actual_js = strip_space(PyDexter()._make_js(chart_type,test_input))
+	assert actual_js == expected_js
+	
+def test_scatter_with_snd_arg():
+        chart_type = 'plot'
+        test_input_x1 = [0,1]
+	test_input_x2 = [0,1]
```
Comment (NathanEpstein) [inline tests/unit.py:17]: indentation is off

File: tests/unit.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	''' compares actual to expected js generated by PyDexter plot method '''
+	test_input = [1,2,3,4,5]
+        chart_type = 'plot'
+        expected_js = "new D3xter({}).plot([1, 2, 3, 4, 5]); \n"
+        actual_js = PyDexter()._make_js(chart_type,test_input)
+        assert actual_js == expected_js
+
+def test_pie():
+	''' compares actual to expected js generated by PyDexter pie method '''
+	test_input = [1,2,3,4,5]
```
Comment (NathanEpstein) [inline tests/unit.py:42]: Indentation again.

File: tests/unit.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	test_input_x1 = [0,1] #use this
+	test_input_x2 = None
+	test_config = {'datasets': [{'y': [0, 1], 'x': [0 ,1]}]}
+	actual_js = PyDexter()._make_js(chart_type,test_config)
+	expected_js = "new D3xter({}).plot({'datasets': [{'y': [0, 1], 'x': [0, 1]}]}); \n"
+	assert actual_js == expected_js
+
+def test_plot():
+	''' compares actual to expected js generated by PyDexter plot method '''
+	test_input = [1,2,3,4,5]
```
Comment (NathanEpstein) [inline tests/unit.py:34]: ditto.

File: tests/unit.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	''' compares actual to expected js generated by PyDexter pie method '''
+	test_input = [1,2,3,4,5]
+        chart_type = 'pie'
+        expected_js = "new D3xter({}).pie([1, 2, 3, 4, 5]); \n"
+        actual_js = PyDexter()._make_js(chart_type,test_input)
+        assert actual_js == expected_js
+
+def test_timeline():
+	''' compares actual to expected js generated by PyDexter timeline method '''
+	test_input = [1,2,3,4,5]
```
Comment (NathanEpstein) [inline tests/unit.py:50]: ditto.

File: tests/unit.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        chart_type = 'timeline'
+        expected_js = "new D3xter({}).timeline([1, 2, 3, 4, 5]); \n"
+        actual_js = PyDexter()._make_js(chart_type,test_input)
+        assert actual_js == expected_js
+
+def test_bar():
+	''' compares actual to expected js generated by PyDexter bar method '''
+	test_input = [1,2,3,4,5]
```
Comment (NathanEpstein) [inline tests/unit.py:58]: ditto.

**Comment by NathanEpstein** [comment]
Minor linting issue - lists alternate between spaced entries and not; `[1,2,3]` versus `[1, 2, 3]`. I'd use spaced entries for consistency.

File: tests/unit.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
+# This file contains a series of unit tests that compare expected and actual javascript code generated by pyd3xter
+# instructions:
+# install pytest: http://pytest.org/latest/
+# $> python -m pytest -v unit.py
+# TODO: refactor to streamline js comparisons; consider edge cases, syntactic curbballs
+# note: expected_js field hand-written
```
Comment (NathanEpstein) [inline tests/unit.py]: Rather than making this note, you can convert `expected_js` to `EXPECTED_JS`. The all-caps is convention for hard-coding.

File: tests/result.txt [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+unit.py::test_hist PASSED
+unit.py::test_scatter_with_snd_arg PASSED
+unit.py::test_scatter_without_snd_arg PASSED
+unit.py::test_plot PASSED
+unit.py::test_pie PASSED
+unit.py::test_timeline PASSED
+unit.py::test_bar PASSED
+
+=========================== 7 passed in 0.09 seconds ===========================
```
Comment (NathanEpstein) [inline tests/result.txt]: This is generated by the testing library you're using right? This file should be added to the `.gitignore`.

**Comment by NathanEpstein** [comment]
Looks good. Comments are all very minor issues. Should be good to ship as soon as they're resolved.

**Comment by ds2643** [comment]
I've prepared the changes you suggested. However, I didn't see any issues with indentation using my text editor-- vim. Do you prefer 4-spaces to 8?

**Comment by ds2643** [comment]
Never mind, found the issue. Will upload later.

File: README.md [hunk 9]
```diff
@@ -114,3 +114,12 @@ pydex.configure({
   'yLab': 'y-axis label'
 })
 ```
+
+### Running tests
+unit.py file contains a series of unit tests that compare expected and actual javascript code generated by pyd3xter
+
+instructions:
+install pytest: http://pytest.org/latest/
+$> python -m pytest -v unit.py
```
Comment (NathanEpstein) [inline README.md:123]: I'm not sure the markdown is rendering the way you intend it to. Maybe try:

```
### Running tests
 unit.py file contains a series of unit tests that compare expected and actual javascript code generated by pyd3xter

* install pytest: http://pytest.org/latest/

* run the tests: `$ python -m pytest -v unit.py`
```

File: tests/unit.py [hunk 10]
```diff
@@ -0,0 +1,62 @@
+from PyDexter import PyDexter
+
+def strip_space(str):
+	return ''.join(str.split())
+
+def test_hist():
+	''' indirectly compares actual to expected js generated by PyDexter hist method '''
+	test_input = [1,2,3,4,5]
```
Comment (NathanEpstein) [inline tests/unit.py:8]: `test_input = [1, 2, 3, 4, 5]`

**Comment by NathanEpstein** [comment]
There are several more of these lists with no spacing. It's minor but it'd be good to be consistent and apply the spaces everywhere.

File: tests/unit.py [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+	chart_type = 'hist'
+	EXPECTED_JS = strip_space("new D3xter({}).hist([1, 2, 3, 4, 5]); \n")
+	actual_js = strip_space(PyDexter()._make_js(chart_type,test_input))
```
Comment (NathanEpstein) [inline tests/unit.py:11]: Same principle, function arguments get spaces throughout the rest of the lib so it'd be good to do in the tests (there are a few cases).

File: tests/unit.py [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+	assert actual_js == EXPECTED_JS
+
+def test_scatter_without_snd_arg():
+	chart_type = 'plot'
+	test_input_x1 = [0,1]
+	test_input_x2 = None
+	test_config = {'datasets': [{'y': [0, 1], 'x': [0 ,1]}]}
+	actual_js = PyDexter()._make_js(chart_type,test_config)
+	EXPECTED_JS = "new D3xter({}).plot({'datasets': [{'y': [0, 1], 'x': [0, 1]}]}); \n"
+	assert actual_js == EXPECTED_JS
```
Comment (NathanEpstein) [inline tests/unit.py:30]: Every test except `test_scatter_with_snd_arg` and `test_scatter_without_snd_arg` has an explanatory comment. I'd either add comments to these two or remove the other comments for consistency.

File: tests/unit.py [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+	test_input = [1,2,3,4,5]
+	chart_type = 'hist'
+	EXPECTED_JS = strip_space("new D3xter({}).hist([1, 2, 3, 4, 5]); \n")
+	actual_js = strip_space(PyDexter()._make_js(chart_type,test_input))
+	assert actual_js == EXPECTED_JS
+	
+def test_scatter_with_snd_arg():
+	chart_type = 'plot'
+	test_input_x1 = [0,1]
+	test_input_x2 = [0,1]
```
Comment (NathanEpstein) [inline tests/unit.py:17]: You're not actually using `test_input_x1` and `test_input_x2` in the code below.

File: tests/unit.py [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+	test_config = {'datasets': [{'y': [0, 1], 'x': [0, 1]}]}
+	actual_js = PyDexter()._make_js(chart_type,test_config)
+	EXPECTED_JS = "new D3xter({}).plot({'datasets': [{'y': [0, 1], 'x': [0, 1]}]}); \n"
+	assert actual_js == EXPECTED_JS
+
+def test_scatter_without_snd_arg():
+	chart_type = 'plot'
+	test_input_x1 = [0,1]
+	test_input_x2 = None
```
Comment (NathanEpstein) [inline tests/unit.py:26]: ditto

**Comment by NathanEpstein** [comment]
Everything here is minor linting stuff and can be addressed in a subsequent PR. This PR is an improvement as is; going to merge.
